js/collections/Moderators.js

```diff
@@ -36,15 +45,17 @@ export default class extends Collection {
 
   add(models, options) {
     let filteredModels = _.isArray(models) ? models : [models];
-    const excludeList = _.pluck(this.excludeCollection.models, 'id');
+    // if an excluded collection was passed into the options, don't add any moderators in it
+    const excludeList = this.options.excludeCollection ?
+        _.pluck(this.options.excludeCollection.models, 'id') : [];
     let modIDs = [];
     // remove any returned profiles that are not valid moderators
     if (models) {
       filteredModels = filteredModels.filter((mod) => {
-        // don't add excluded ids
-        const notExcluded = excludeList.indexOf(mod.id) === -1;
+        // don't add excluded ids or your own id
+        const notExcluded = excludeList.indexOf(mod.id) === -1 && mod.id !== app.profile.id;
         // don't add if not a mod or the mod data is missing
-        return mod.moderator && mod.modInfo && notExcluded;
+        return mod.get('moderator') && mod.get('modInfo') && notExcluded;
```

Comment:
The profile model has a 'isModerator' getter which would handle the first two conditions.